snooty/language_server.py

```diff
@@ -298,6 +299,24 @@ def m_text_document__get_page_ast(self, filePath: str) -> SerializableType:
 
         return self.project.get_page_ast(Path(filePath))
 
+    def m_text_document__get_project_name(self) -> SerializableType:
+        """Get the project's name from its snooty.toml file"""
+        if not self.project:
+            logger.warn("Project uninitialized")
+            return None
+
+        return self.project.get_project_name()
+
+    def m_text_document__get_page_fileid(self, filePath: str) -> SerializableType:
+        """Given a path to a file, return its fileid as a string"""
+        if not self.project:
+            logger.warn("Project uninitialized")
+            return None
+
+        fileid = self.project.get_fileid(PurePath(filePath))
+        fileid = fileid.with_name(PAT_FILE_EXTENSIONS.sub("", fileid.name))
```

Comment:
I'd almost rather that this be a `without_known_suffix()` property method on `FileId`; as-is it takes a second to understand the logic.